src/utilities.ts

```diff
@@ -113,6 +114,25 @@ const formatBytes = (bytes: number, decimals = 2) => {
   return `${count} ${unit}`;
 };
 
+const verifyScanStatus = (scanStatus: ScanStatus): boolean => {
+  if (scanStatus === ScanStatus.FailedWithIssues) {
+    soosLogger.info("Analysis complete - Failures reported");
+    soosLogger.info("Failing the build.");
+    return true;
+  } else if (scanStatus === ScanStatus.Incomplete) {
+    soosLogger.info(
+      "Analysis Incomplete. It may have been cancelled or superseded by another scan.",
+    );
+    soosLogger.info("Failing the build.");
+    return true;
+  } else if (scanStatus === ScanStatus.Error) {
+    soosLogger.info("Analysis Error.");
+    soosLogger.info("Failing the build.");
+    return true;
+  }
+  return false;
+};
```

Comment:
```suggestion
const verifyScanStatus = (scanStatus: ScanStatus): boolean => {
  let fail = false;
  if (scanStatus === ScanStatus.FailedWithIssues) {
    soosLogger.warn("Analysis complete - Failures reported");
    fail = true;
  } else if (scanStatus === ScanStatus.Incomplete) {
    soosLogger.warn(
      "Analysis Incomplete. It may have been cancelled or superseded by another scan.",
    );
    fail = true;
  } else if (scanStatus === ScanStatus.Error) {
    soosLogger.warn("Analysis Error.");
    fail = true;
  }
  
  if (fail) {
    soosLogger.warn("Failing the build.");  
  }
  
  return fail;
};
```